k4Reco/Overlay/components/OverlayTiming.h

```diff
@@ -0,0 +1,225 @@
+                             KeyValues("OutputMCParticles", {"NewMCParticles"}),
+                             KeyValues("OutputSimTrackerHits", {"MCParticles1"}),
+                             KeyValues("OutputSimCalorimeterHits", {"MCParticles2"}),
+                             KeyValues("OutputCaloHitContributions", {"OverlayCaloHitContributions"})
+                         }) {}
+
+  template <typename T> void overlayCollection(std::string collName, const podio::CollectionBase& inColl);
+
+  virtual StatusCode initialize() final;
+  /**  Execute.
+   *   @return status code
+   */
+  virtual StatusCode finalize() final;
+
+  retType virtual operator()(
+      const edm4hep::EventHeaderCollection& headers, const edm4hep::MCParticleCollection& mcParticles,
+      const std::map<std::string, const edm4hep::SimTrackerHitCollection&>&     simTrackerHits,
+      const std::map<std::string, const edm4hep::SimCalorimeterHitCollection&>& simCalorimeterHits,
+      const edm4hep::CaloHitContributionCollection&                             caloHitContribs) const final;
+
+  std::tuple<float, float, bool> define_time_windows(const std::string& Collection_name) const;
+
+private:
+  Gaudi::Property<bool>        _randomBX{this, "RandomBx", false,
+                                  "Place the physics event at an random position in the train: overrides PhysicsBX"};
+  mutable Gaudi::Property<int> _BX_phys{this, "PhysicsBX", 1, "Number of the Bunch crossing of the physics event"};
+  Gaudi::Property<float>       _tpcVdrift_mm_ns{this, "TPCDriftvelocity", float(5.0e-2),
+                                          "Drift velocity in TPC [mm/ns] - default 5.0e-2 (5cm/us)"};
+  Gaudi::Property<int>         _nBunchTrain{this, "NBunchtrain", 1, "Number of bunches in a bunch train"};
+  mutable Gaudi::Property<int> m_startWithBackgroundFile{this, "StartBackgroundFileIndex", -1,
+                                                         "Which background file to startWith"};
+  Gaudi::Property<int>         m_startWithBackgroundEvent{this, "StartBackgroundEventIndex", -1,
+                                                  "Which background event to startWith"};
+
+  Gaudi::Property<std::vector<std::vector<std::string>>> m_inputFileNames{
+      this,
+      "BackgroundFileNames",
+      {},
+      "Name of the lcio input file(s) with background - assume one file per bunch crossing."};
+
+  Gaudi::Property<double> _NOverlay{this, "NumberBackground", 1,
+                                    "Number of Background events to overlay - either fixed or Poisson mean"};
+
+  Gaudi::Property<bool> _Poisson{
+      this, "Poisson_random_NOverlay", false,
+      "Draw random number of Events to overlay from Poisson distribution with  mean value NumberBackground"};
+
+  Gaudi::Property<std::string> _mcParticleCollectionName{this, "MCParticleCollectionName", "MCParticle",
+                                                         "The MC Particle Collection Name"};
```

Comment:
```suggestion
  Gaudi::Property<std::string> _mcParticleCollectionName{this, "MCParticleCollectionName", "MCParticle",
                                                         "The name of the MCParticle collection in the background files"};
```
maybe clearer like this? :)